lib/web_ui/lib/src/engine/view_embedder/dimensions_provider/custom_element_dimensions_provider.dart

```diff
@@ -15,18 +17,27 @@ import 'dimensions_provider.dart';
 /// All the measurements returned from this class are potentially *expensive*,
 /// and should be cached as needed. Every call to every method on this class
 /// WILL perform actual DOM measurements.
+///
+/// This broadcasts `null` size events, to match the implementation of the
+/// FullPageDimensionsProvider, but it could broadcast the size coming from the
+/// DomResizeObserverEntry. Further changes in the engine are required for this
+/// to be effective.
```

Comment:
Why does `FullPageDimensionsProvider` return null? I also read the docs on `DimensionsProvider`, and also failed to understand what null means:

```dart
  /// Returns a Stream with the changes to [ui.Size] (when cheap to get).
  Stream<ui.Size?> get onResize;
```

Reply:
FullPageDimensionsProvider returns null because the `onresize` event we're using does NOT broadcast the new size of the window, only that it changed (unless we trigger a re-measurement).

Since triggering the re-measurement seemed to be expensive (that behavior is inherited from the old code), we ended up with:

* The FullPage uses the resize stream as if it were a `Stream<void>` and just broadcasts `null` events, and everywhere else that listens for it then goes on and re-computes the physical size anyways.
* The CustomElement DOES have the new size, so it used to be a `Stream<Size>`, however, since the code always assumes that the size is unavailable, it still ends up recomputing the physical size.

This change just makes the CustomElement dimensions provider "equivalent" to the FullScreen one.

PS: In the future, we should eventually stop triggering re-measurements of the containers (even in full-screen), so we can use the "last seen Size" as the physical size of the app.

Reply:
Ah, cool. I'd just copy'n'paste this explanation into the dartdoc for the abstract `onResize` getter.

Reply:
Added to the docs!